#### PART TWO (DATA ANALYSIS)
Instructions: • Use any programming language of your choice to analyze data provided in the “wine_dataset.csv” file and answer the following questions and submit your response in PDF:

Q1 Which wine type, red or white, has the highest average (mean) pH?

Q2 What is the average alcohol content for wines that scored more than 6 on quality?

Q3 Looking only at white wines, find the average quality for wines with volatile_acidity greater than or equal to 0.5, wines with volatile_acidity under 0.5 and wines with no volatile_acidity information?

Q4 Find the ratio of average alcohol in white wines to that of red wines for each quality rating?

In [1]:
getwd()

[1] "C:/Users/EDEH EMEKA NWEKE"

In [6]:
library (dplyr)


Attaching package: 'dplyr'

The following objects are masked from 'package:stats':

    filter, lag

The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union



In [4]:
wine <- read.csv("wine_dataset.csv")
head(wine)

ID type  fixed_acidity volatile_acidity citric_acid residual_sugar chlorides
1 1  white 7.0           0.27             0.36        20.7           0.045    
2 2  white 6.3           0.30             0.34         1.6           0.049    
3 3  white 8.1           0.28             0.40         6.9           0.050    
4 4  white 7.2           0.23             0.32         8.5           0.058    
5 5  white 7.2           0.23             0.32         8.5           0.058    
6 6  white 8.1           0.28             0.40         6.9           0.050    
  free_sulfur_dioxide total_sulfur_dioxide density pH   sulphates alcohol
1 45                  170                  1.0010  3.00 0.45       8.8   
2 14                  132                  0.9940  3.30 0.49       9.5   
3 30                   97                  0.9951  3.26 0.44      10.1   
4 47                  186                  0.9956  3.19 0.40       9.9   
5 47                  186                  0.9956  3.19 0.40       9.9   
6 30                   97                  0.9951  3.26 0.44      10.1   
  quality
1 6      
2 6      
3 6      
4 6      
5 6      
6 6

#### Q1 Which wine type, red or white, has the highest average (mean) pH?

In [7]:
Ans1 <-wine %>% 
      select(type, pH) %>%
      group_by(type) %>% 
      summarize(wine_type_avg=mean(pH,na.rm=TRUE))

Ans1

type  wine_type_avg
1 red   3.310864     
2 white 3.188203

From the above result, it's evident that red wine, with average pH value of 3.310864, has a higher average pH.

#### Q2 What is the average alcohol content for wines that scored more than 6 on quality?

In [8]:
Ans2 <- wine %>%
      filter(quality > 6)%>% 
      summarise(avg_alcohol = mean(alcohol, na.rm=TRUE))

Ans2

avg_alcohol
1 11.43336

#### Q3 Looking only at white wines, find the average quality for wines with volatile_acidity greater than or equal to 0.5, wines with volatile_acidity under 0.5 and wines with no volatile_acidity information?

In [9]:
#Average quality for wines with volatile_acidity greater than or equal to 0.5

Avg_quality_greater_or_equal <- wine %>%
      filter(type == 'white', volatile_acidity >= 0.5)%>% 
      summarise(avg_quality = mean(quality, na.rm=TRUE))

Avg_quality_greater_or_equal

avg_quality
1 5.276471

In [10]:
#Average quality for wines with volatile_acidity under 0.5

Avg_quality_under <- wine %>%
      filter(type == 'white', volatile_acidity < 0.5)%>% 
      summarise(avg_quality_under = mean(quality, na.rm=TRUE))

Avg_quality_under

avg_quality_under
1 5.899386

In [11]:
#Average quality for wines with no volatile_acidity information

Avg_quality_with_no <- wine %>%
      filter(type == 'white', volatile_acidity == 0)%>% 
      summarise(avg_quality_with_no_acidity = mean(quality, na.rm=TRUE))

Avg_quality_with_no

avg_quality_with_no_acidity
1 NaN

#### Q4 Find the ratio of average alcohol in white wines to that of red wines for each quality rating?

In [ ]:
# I am yet to get this right

for (i in wine$quality) {
      wine %>%
      (filter(type == 'white')%>% 
      select(type, quality)%>%
      summarise(avg_alc = mean(wine$alcohol, na.rm=TRUE)))%>%
         print()
    print(i)  }
